everest/simulator/everest2res.py

```diff
@@ -347,9 +347,9 @@ def _is_dir_all_geo(source, ever_config: EverestConfig):
 
 
 def _extract_data_operations(ever_config: EverestConfig):
-    symlink_fmt = "symlink {source} {link_name}"
-    copy_dir_fmt = "copy_directory {source} {target}"
-    copy_file_fmt = "copy_file {source} {target}"
+    symlink_fmt = "SYMLINK {source} {link_name}"
```

Comment:
Did some investigation and it will be fixed on ERT side. Turns out the lower case were all installed via `INSTALL_JOB_DIRECTORY` in site config, and they show up on the ert-side always without arglist and required args, so basically no argument validation. And some Everest tests expects this behavior, so I will add something on the ERT side to create the lower case included steps without arg validation. 

![Screenshot 2024-07-09 at 08 39 43](https://github.com/equinor/everest/assets/11595637/1ac792c6-8df4-4a16-b003-4610681d72fe)

Textual diff
```

 'symlink': ForwardModelStep(name='symlink',                     executable='ert/src/ert/shared/share/ert/shell_scripts/symlink.py', stdin_file=None, stdout_file='symlink.stdout', stderr_file='symlink.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'SYMLINK': ForwardModelStep(name='SYMLINK',                     executable='ert/src/ert/shared/share/ert/shell_scripts/symlink.py', stdin_file=None, stdout_file='SYMLINK.stdout', stderr_file='SYMLINK.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<TARGET>', '<LINKNAME>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'ECLIPSE300': ForwardModelStep(name='ECLIPSE300',               executable='ert/src/ert/shared/share/ert/forward-models/res/script/ecl300.py', stdin_file=None, stdout_file='ECLIPSE300.stdout', stderr_file='ECLIPSE300.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=['<ECLBASE>', '-v', '<VERSION>', '-n', '<NUM_CPU>', '<OPTS>'], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={'<VERSION>': 'version', '<NUM_CPU>': '1', '<OPTS>': ''}, private_args=<SubstitutionList([])>, help_text=''),
 'eclipse300': ForwardModelStep(name='eclipse300',               executable='ert/src/ert/shared/share/ert/forward-models/res/script/ecl300.py', stdin_file=None, stdout_file='eclipse300.stdout', stderr_file='eclipse300.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'MAKE_DIRECTORY': ForwardModelStep(name='MAKE_DIRECTORY',       executable='ert/src/ert/shared/share/ert/shell_scripts/make_directory.py', stdin_file=None, stdout_file='MAKE_DIRECTORY.stdout', stderr_file='MAKE_DIRECTORY.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=1, max_arg=1, arglist=['<DIRECTORY>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'make_directory': ForwardModelStep(name='make_directory',       executable='ert/src/ert/shared/share/ert/shell_scripts/make_directory.py', stdin_file=None, stdout_file='make_directory.stdout', stderr_file='make_directory.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
  
 'DELETE_DIRECTORY': ForwardModelStep(name='DELETE_DIRECTORY',   executable='ert/src/ert/shared/share/ert/shell_scripts/delete_directory.py', stdin_file=None, stdout_file='DELETE_DIRECTORY.stdout', stderr_file='DELETE_DIRECTORY.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=1, max_arg=1, arglist=['<DIRECTORY>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'delete_directory': ForwardModelStep(name='delete_directory',   executable='ert/src/ert/shared/share/ert/shell_scripts/delete_directory.py', stdin_file=None, stdout_file='delete_directory.stdout', stderr_file='delete_directory.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),

 'MAKE_SYMLINK': ForwardModelStep(name='MAKE_SYMLINK',           executable='ert/src/ert/shared/share/ert/shell_scripts/symlink.py', stdin_file=None, stdout_file='MAKE_SYMLINK.stdout', stderr_file='MAKE_SYMLINK.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<TARGET>', '<LINKNAME>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'make_symlink': ForwardModelStep(name='make_symlink',           executable='ert/src/ert/shared/share/ert/shell_scripts/symlink.py', stdin_file=None, stdout_file='make_symlink.stdout', stderr_file='make_symlink.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'CAREFUL_COPY_FILE': ForwardModelStep(name='CAREFUL_COPY_FILE', executable='ert/src/ert/shared/share/ert/shell_scripts/careful_copy_file.py', stdin_file=None, stdout_file='CAREFUL_COPY_FILE.stdout', stderr_file='CAREFUL_COPY_FILE.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<FROM>', '<TO>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'careful_copy_file': ForwardModelStep(name='careful_copy_file', executable='ert/src/ert/shared/share/ert/shell_scripts/careful_copy_file.py', stdin_file=None, stdout_file='careful_copy_file.stdout', stderr_file='careful_copy_file.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'ECLIPSE100': ForwardModelStep(name='ECLIPSE100',               executable='ert/src/ert/shared/share/ert/forward-models/res/script/ecl100.py', stdin_file=None, stdout_file='ECLIPSE100.stdout', stderr_file='ECLIPSE100.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=['<ECLBASE>', '-v', '<VERSION>', '-n', '<NUM_CPU>', '<OPTS>'], required_keywords=['<VERSION>'], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={'<NUM_CPU>': '1', '<OPTS>': ''}, private_args=<SubstitutionList([])>, help_text=''),
 'eclipse100': ForwardModelStep(name='eclipse100',               executable='ert/src/ert/shared/share/ert/forward-models/res/script/ecl100.py', stdin_file=None, stdout_file='eclipse100.stdout', stderr_file='eclipse100.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'TEMPLATE_RENDER': ForwardModelStep(name='TEMPLATE_RENDER',     executable='ert/src/ert/shared/share/ert/forward-models/templating/script/template_render.py', stdin_file=None, stdout_file='TEMPLATE_RENDER.stdout', stderr_file='TEMPLATE_RENDER.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=['-i', '<INPUT_FILES>', '-o', '<OUTPUT_FILE>', '-t', '<TEMPLATE_FILE>'], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'template_render': ForwardModelStep(name='template_render',     executable='ert/src/ert/shared/share/ert/forward-models/templating/script/template_render.py', stdin_file=None, stdout_file='template_render.stdout', stderr_file='template_render.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'DELETE_FILE': ForwardModelStep(name='DELETE_FILE',             executable='ert/src/ert/shared/share/ert/shell_scripts/delete_file.py', stdin_file=None, stdout_file='DELETE_FILE.stdout', stderr_file='DELETE_FILE.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=1, max_arg=1, arglist=['<FILES>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'delete_file': ForwardModelStep(name='delete_file',             executable='ert/src/ert/shared/share/ert/shell_scripts/delete_file.py', stdin_file=None, stdout_file='delete_file.stdout', stderr_file='delete_file.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'COPY_DIRECTORY': ForwardModelStep(name='COPY_DIRECTORY',       executable='ert/src/ert/shared/share/ert/shell_scripts/copy_directory.py', stdin_file=None, stdout_file='COPY_DIRECTORY.stdout', stderr_file='COPY_DIRECTORY.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<FROM>', '<TO>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'copy_directory': ForwardModelStep(name='copy_directory',       executable='ert/src/ert/shared/share/ert/shell_scripts/copy_directory.py', stdin_file=None, stdout_file='copy_directory.stdout', stderr_file='copy_directory.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),

 'COPY_FILE': ForwardModelStep(name='COPY_FILE',                 executable='ert/src/ert/shared/share/ert/shell_scripts/copy_file.py', stdin_file=None, stdout_file='COPY_FILE.stdout', stderr_file='COPY_FILE.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<FROM>', '<TO>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'copy_file': ForwardModelStep(name='copy_file',                 executable='ert/src/ert/shared/share/ert/shell_scripts/copy_file.py', stdin_file=None, stdout_file='copy_file.stdout', stderr_file='copy_file.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),

 'MOVE_FILE': ForwardModelStep(name='MOVE_FILE',                 executable='ert/src/ert/shared/share/ert/shell_scripts/move_file.py', stdin_file=None, stdout_file='MOVE_FILE.stdout', stderr_file='MOVE_FILE.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=2, max_arg=2, arglist=['<FROM>', '<TO>'], required_keywords=[], arg_types=[<SchemaItemType.STRING: 'STRING'>, <SchemaItemType.STRING: 'STRING'>], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 'move_file': ForwardModelStep(name='move_file',                 executable='ert/src/ert/shared/share/ert/shell_scripts/move_file.py', stdin_file=None, stdout_file='move_file.stdout', stderr_file='move_file.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),
 
 'FLOW': ForwardModelStep(name='FLOW',                           executable='ert/src/ert/shared/share/ert/forward-models/res/script/flow.py', stdin_file=None, stdout_file='FLOW.stdout', stderr_file='FLOW.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=['<ECLBASE>', '-v', '<VERSION>', '-n', '<NUM_CPU>', '<OPTS>'], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={'<VERSION>': 'default', '<NUM_CPU>': '1', '<OPTS>': ''}, private_args=<SubstitutionList([])>, help_text='')}
 'flow': ForwardModelStep(name='flow',                           executable='ert/src/ert/shared/share/ert/forward-models/res/script/flow.py', stdin_file=None, stdout_file='flow.stdout', stderr_file='flow.stderr', start_file=None, target_file=None, error_file=None, max_running_minutes=None, min_arg=None, max_arg=None, arglist=[], required_keywords=[], arg_types=[], environment={'_ERT_ITERATION_NUMBER': '<ITER>', '_ERT_REALIZATION_NUMBER': '<IENS>', '_ERT_RUNPATH': '<RUNPATH>'}, exec_env={}, default_mapping={}, private_args=<SubstitutionList([])>, help_text=''),

```